client/jetpack-cloud/sections/partner-portal/partner-portal-checkbox/index.tsx

```diff
@@ -0,0 +1,19 @@
+import { Gridicon } from '@automattic/components';
+import classNames from 'classnames';
+import { ReactElement } from 'react';
+import './style.scss';
+
+interface Props {
+	isChecked: boolean;
+}
+
+export default function PartnerPortalCheckbox( props: Props ): ReactElement {
+	const { isChecked } = props;
+
+	const classes = classNames( {
+		'partner-portal-checkbox': true,
+		'is-checked': isChecked,
+	} );
+
+	return <div className={ classes }>{ isChecked && <Gridicon icon="checkmark" /> }</div>;
```

Comment:
Maybe adding `role="checkbox" aria-checked={ isChecked }` could improve the experience and accessibility.

Considering we are replacing `FormRadio`, which implements an `input` tag of `type="radio"`,  with `PartnerPortalCheckbox`, we might even consider making/naming this off of "radio" instead of "checkbox", since it behaves like a radio button and not like a checkbox.

I would go with checkbox and adding the aria attributes at least, for scope trimming.